occ_poll_app_handler.cpp

```diff
@@ -0,0 +1,556 @@
+                    (static_cast<uint32_t>(PollRspData[index + 9]));
+
+            PollRspSoftMin = (static_cast<uint32_t>(PollRspData[index + 10]) << 8) |
+                    (static_cast<uint32_t>(PollRspData[index + 11]));
+
+            index += bytesPerSensor;
+        }
+    }
+    else
+    {
+        if (TraceOncePollSensor)
+        {
+            lg2::error("OccPollAppHandler::PushCapsSensorFormat: unsuported format :{BYTE}",
+                "BYTE", lg2::hex, SensorFormat);
+            TraceOncePollSensor = false;
+        }
+
+            index += bytesPerSensor * NumberSensors;
+    }
+
+
+
+}//end PushCapsSensorsToDbus
+
+void OccPollAppHandler::PushExtnSensorsToDbus(uint16_t& index )
+{
+    // Extn Sensor Header Data format for POLL
+    //  SensorFormat: 1 byte
+    //  bytesPerSensor: 1 byte
+    //  NumberSensors: 1 byte
+    //----------------------
+    uint8_t SensorFormat = PollRspData[index++];
+    uint8_t bytesPerSensor = PollRspData[index++];
+    uint8_t NumberSensors = PollRspData[index++];
+
+    if (SensorFormat == EXTN_SENSOR_FORMAT)
+    {
+        for (uint16_t i = 0; i < NumberSensors; i++)
+        {
+            // Extn Sensor Record Data format for POLL
+            //  SensorID:  4 byte
+            //  Flags:     1 byte
+            //  Reserved:  1 byte
+            //  Value:     6 byte
+            //                 PWRM and PWRP last 2 bytes.
+            //----------------------
+
+            const uint32_t SensorName = (static_cast<uint32_t>(PollRspData[index]) << 24) |
+                    (static_cast<uint32_t>(PollRspData[index + 1]) << 16) |
+                    (static_cast<uint32_t>(PollRspData[index + 2]) << 8) |
+                    (static_cast<uint32_t>(PollRspData[index + 3]));
+
+            bool push_power_to_dbus = false;
+            std::string sensorPath = OCC_SENSORS_ROOT;
+            uint32_t SensorValue = 0;
+
+            switch (SensorName)
+            {
+                case EXTN_LABEL_FMIN: break;
+                case EXTN_LABEL_FDIS: break;
+                case EXTN_LABEL_FBAS: break;
+                case EXTN_LABEL_FUT:  break;
+                case EXTN_LABEL_FMAX: break;
+                case EXTN_LABEL_CLIP: break;
+                case EXTN_LABEL_MODE: break;
+                case EXTN_LABEL_WOFC: break;
+                case EXTN_LABEL_WOFI: break;
+                case EXTN_LABEL_PWRM:
+                    push_power_to_dbus = true;
+                    sensorPath.append("/power/chiplet" + std::to_string(occInstanceID) + "_mem_power");
+                    break;
+                case EXTN_LABEL_PWRP:
+                    push_power_to_dbus = true;
+                    sensorPath.append("/power/chiplet" + std::to_string(occInstanceID) + "_power");
+                    break;
+                case EXTN_LABEL_ERRH: break;
+                default:
+                    lg2::error("OccPollAppHandler::PushExtnSensorsToDbus: EXTN label name FAILED:{BYTE}",
```

Comment:
probably skip tracing.  If OCC added a new label, this would get traced on every poll command.
We really only care about the ones we need.